app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/vm.rb

```diff
@@ -112,15 +112,69 @@ def raw_destroy
     unsupported_reason_add(:resize, _('The VM is not powered off')) unless current_state == "off"
   end
 
-  def raw_resize(new_flavor)
+  def raw_resize(options)
     with_provider_object do |instance|
-      instance.resize(instance, new_flavor.name)
+      instance.resize(instance, options["flavor"])
     rescue IBMCloudSdkCore::ApiException => e
       Notification.create(:type => :vm_resize_error, :options => {:subject => instance[:name], :error => e.error})
       raise MiqException::MiqProvisionError, e.to_s
     end
   end
 
+  def params_for_resize
+    {
+      :fields => [
+        {
+          :component  => 'select',
+          :name       => 'flavor',
+          :id         => 'flavor',
+          :label      => _('Flavor'),
+          :isRequired => true,
+          :options    => resize_form_options.map do |flavor|
+            {
+              :label => flavor[:name],
+              :value => flavor[:id],
+            }
+          end,
```

Comment:
If we're going to build an array of hashes down here https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/420/files#diff-5af23959f8142218f0266befea8e232991173bc7a177a5b6a24154a782cc2c2eR150 we might as well make it in the format that is being expected here:
```suggestion
          :options    => resize_form_options,
```